services/storage/src/simcore_service_storage/handlers.py

```diff
@@ -267,25 +269,49 @@ async def _go():
         return {"error": None, "data": sync_results}
 
 
-# DISABLED: @routes.patch(f"/{api_vtag}/locations/{{location_id}}/files/{{fileId}}/metadata") # type: ignore
+@routes.patch(f"/{api_vtag}/locations/{{location_id}}/files/{{fileId}}/metadata")  # type: ignore
 async def update_file_meta_data(request: web.Request):
     params, query, body = await extract_and_validate(request)
 
     assert params, "params %s" % params  # nosec
     assert query, "query %s" % query  # nosec
     assert not body, "body %s" % body  # nosec
 
-    assert params["location_id"]  # nosec
-    assert params["fileId"]  # nosec
-    assert query["user_id"]  # nosec
+    with handle_storage_errors():
+        file_uuid = urllib.parse.unquote_plus(params["fileId"])
+        user_id = query["user_id"]
+
+        dsm = await _prepare_storage_manager(params, query, request)
+
+        data: Optional[FileMetaDataEx] = await dsm.update_metadata(
+            file_uuid=file_uuid, user_id=user_id
+        )
+        if data is None:
+            raise web.HTTPNotFound(reason=f"Could not update metadata for {file_uuid}")
+
+        return {
+            "error": None,
+            "data": {**attr.asdict(data.fmd), "parent_id": data.parent_id},
+        }
+
+
+@routes.delete(f"/{api_vtag}/locations/{{location_id}}/files/{{fileId}}/metadata")  # type: ignore
+async def delete_file_meta_data(request: web.Request):
+    params, query, body = await extract_and_validate(request)
+
+    assert params, "params %s" % params  # nosec
+    assert query, "query %s" % query  # nosec
+    assert not body, "body %s" % body  # nosec
 
     with handle_storage_errors():
-        location_id = params["location_id"]
-        _user_id = query["user_id"]
-        _file_uuid = params["fileId"]
+        user_id = query["user_id"]
+        file_uuid = urllib.parse.unquote_plus(params["fileId"])
 
         dsm = await _prepare_storage_manager(params, query, request)
-        _location = dsm.location_from_id(location_id)
+
+        await dsm.delete_metadata(user_id=user_id, file_uuid=file_uuid)
+
+    return web.HTTPNoContent(content_type="application/json")
```

Comment:
You typically raise ``web.HTTP...`` even if there are not failures. Those are both exceptions and responses https://docs.aiohttp.org/en/latest/web_exceptions.html

Reply:
raise apparently is the way to go.